pip/commands/list.py

```diff
@@ -146,15 +153,20 @@ def run(self, options, args):
             raise CommandError(
                 "Options --outdated and --uptodate cannot be combined.")
 
+        # TODO (nvdv): Check options combinations.
+
```

Comment:
As noted, checks that incompatible combinations of arguments aren't given need to be added.

In theory, I guess ```--nodeps``` could be used with many of the other options, but I'd be inclined to say YAGNI for now.